### PR TITLE
Incorporate touchscreen in robot

### PR DESCRIPTION
**Pull request by Lu-ni**
- integrate touch screen driver
- create a folder for all the gui dependencies 
- create a file by page on the touch screen
- polymorphism !
- Remove log/score estimation to show a text and a button

### CODE REVIEW

File: master-firmware/src/gui.c [hunk 1]
```diff
@@ -6,72 +6,122 @@
...
 #define MSG_MAX_LENGTH 128
-#define MSG_BUF_SIZE   16
+#define MSG_BUF_SIZE 16
+
 static char msg_buffer[MSG_MAX_LENGTH][MSG_BUF_SIZE];
 static char *msg_mailbox_buf[MSG_BUF_SIZE];
 static MAILBOX_DECL(msg_mailbox, msg_mailbox_buf, MSG_BUF_SIZE);
 static MEMORYPOOL_DECL(msg_pool, MSG_MAX_LENGTH, NULL);
 
+float calibrationData[] = {
```
Comment (antoinealb) [inline master-firmware/src/gui.c]: Should we store this in each robot's config maybe ?

File: master-firmware/src/gui.c [hunk 2]
```diff
@@ -6,72 +6,122 @@
...
+    };
+    menu_t my_menu = {pages, sizeof(pages) / sizeof(page_t*)};
+    menu_initialize(&my_menu);
+
     chPoolLoadArray(&msg_pool, msg_buffer, MSG_BUF_SIZE);
     init_done = true;
 
     WARNING("GUI init done");
 
+    NOTICE("test %d", 42);
```
Comment (antoinealb) [inline master-firmware/src/gui.c]: Remove test code

File: master-firmware/src/main.c [hunk 3]
```diff
@@ -228,25 +228,6 @@ int main(void)
                         MAX_NB_BUS_ENUMERATOR_ENTRIES);
 
 
-    static __attribute__((section(".ccm"))) motor_driver_t motor_driver_buffer[MAX_NB_MOTOR_DRIVERS];
```
Comment (antoinealb) [inline master-firmware/src/main.c]: We still need this code ! ;)

File: master-firmware/src/gui.c (hunk 2 above)
Comment (Lu-ni) [reply master-firmware/src/gui.c]: done

File: master-firmware/src/adafruit_display_driver/gmouse_lld_STMPE610_board.h [hunk 4]
```diff
@@ -0,0 +1,89 @@
+/*
```
[the remark this answers is not in the document]
Comment (antoinealb) [reply master-firmware/src/adafruit_display_driver/gmouse_lld_STMPE610_board.h]: it *is* used by the uGFX driver implementaiton, AFAIK

File: master-firmware/src/adafruit_display_driver/board_hx8357.c [hunk 5]
```diff
@@ -13,29 +13,32 @@
 
 void hx8357_init_board(void *g)
 {
-    (void) g;
+    (void)g;
     static SPIConfig spi_cfg = {
         .end_cb = NULL,
-        .cr1 = SPI_CR1_BR_1,
+        // .cr1 = SPI_CR1_BR_1 | SPI_CR1_BR_0, // div 16, 5.6 MHz
```
[the remark this answers is not in the document]
Comment (Lu-ni) [reply master-firmware/src/adafruit_display_driver/board_hx8357.c]: I think I removed all dead code

File: master-firmware/src/gui.c (hunk 1 above)
Comment (Lu-ni) [reply master-firmware/src/gui.c]: You'll have to show me how this work, I actually don't have any clue about it

File: master-firmware/src/main.c (hunk 3 above)
Comment (Lu-ni) [reply master-firmware/src/main.c]: It's back, with other thing, reverted the commit where we "disable some stuff"

File: master-firmware/src/gui.c [hunk 6]
```diff
@@ -84,14 +134,20 @@ void gui_start()
 void gui_log_console(struct error *e, va_list args)
 {
     static char buffer[256];
-    if (init_done) {
+    return;
```
[the remark this answers is not in the document]
Comment (Lu-ni) [reply master-firmware/src/gui.c]: What do you mean ? is this wrong ?

File: master-firmware/src/commands.c [hunk 7]
```diff
@@ -1595,6 +1595,28 @@ static void cmd_lever_full(BaseSequentialStream *chp, int argc, char *argv[])
     chprintf(chp, "right lever: %d\r\n", strat_lever_is_full(LEVER_SIDE_RIGHT));
 }
 
+
+static void cmd_touchscreen(BaseSequentialStream *chp, int argc, char *argv[])
```
[the remark this answers is not in the document]
Comment (Lu-ni) [reply master-firmware/src/commands.c]: done

File: master-firmware/src/adafruit_display_driver/board_hx8357.c [hunk 8]
```diff
@@ -57,18 +60,20 @@ static uint8_t cache[CACHE_SIZE];
...
 void hx8357_write_cache(GDisplay *g, uint16_t c)
 {
-    cache[cache_depth ++] = c >> 8;
-    cache[cache_depth ++] = c & 0xff;
+    cache[cache_depth++] = c >> 8;
+    cache[cache_depth++] = c & 0xff;
 
-    if (cache_depth == CACHE_SIZE) {
+    if (cache_depth == CACHE_SIZE)
+    {
```
[the remark this answers is not in the document]
Comment (Lu-ni) [reply master-firmware/src/adafruit_display_driver/board_hx8357.c]: Is it better  now ? i used your tool on each file we use for the gui

[the remark this answers is not in the document]
Comment (Lu-ni) [reply master-firmware/src/adafruit_display_driver/board_hx8357.c]: btw, i got this error while using uncrustify:
/home/lucas/src/cvra/coding-style/uncrustify.cfg:720 Unknown symbol 'align_number_left'

**Pull request merged**